### PR TITLE
Goal end date is no longer a required field

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Make a goal's end date an optional field. Also added some tests for front end test coverage.
In addition, fixed a bug that made it unreasonably hard to get read only resources

## How to test
Add an end date to the create goals form, or not. Try it both ways, then do the same on the Activity Report.
Confirm that resources on the RTR are now read only if a goal is closed OR and objective is completed

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1072

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/components/GoalForm/ResourceRepeater.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ export default function ResourceRepeater({
         <p className="usa-prose text-bold margin-bottom-0">Resource links</p>
         <ul className="usa-list usa-list--unstyled">
           {onlyResourcesWithValues.map((resource) => (
-            !(status === 'Complete' && goalStatus === 'Closed') || resource.onAnyReport ? (
+            !(status === 'Complete' || goalStatus === 'Closed') || resource.onAnyReport ? (
```
Comment (AdamAdHocTeam) [inline frontend/src/components/GoalForm/ResourceRepeater.js:41]: Not sure if this should be mentioned in testing.

File: frontend/src/components/__tests__/AutomaticResizingTextarea.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import AutomaticResizingTextarea from '../AutomaticResizingTextarea';
+
+describe('AutomaticResizingTextarea', () => {
```
Comment (AdamAdHocTeam) [inline frontend/src/components/__tests__/AutomaticResizingTextarea.js:6]: 📖

File: frontend/src/components/GoalForm/ResourceRepeater.js (hunk 1 above)
Comment (thewatermethod) [reply frontend/src/components/GoalForm/ResourceRepeater.js:41]: do you mean in the PR notes?

Comment (AdamAdHocTeam) [reply frontend/src/components/GoalForm/ResourceRepeater.js:41]: Yea just for testing purposes.

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good was able to create goals with and without end dates (rtr/ar).

**Pull request merged**